src/Features/CSharp/Portable/CodeRefactorings/ConvertToInterpolatedString/ConvertToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Simplification;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertToInterpolatedString
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertToInterpolatedString), Shared]
+    internal partial class ConvertToInterpolatedStringRefactoringProvider : AbstractConvertToInterpolatedStringRefactoringProvider<InterpolatedStringExpressionSyntax, InvocationExpressionSyntax, ExpressionSyntax, ArgumentSyntax, LiteralExpressionSyntax>
+    {
+        protected override SeparatedSyntaxList<ArgumentSyntax>? GetArguments(InvocationExpressionSyntax invocation) =>
+            invocation?.ArgumentList?.Arguments;
+
+        protected override ImmutableArray<ExpressionSyntax> GetExpandedArguments(
+            SemanticModel semanticModel, 
+            SeparatedSyntaxList<ArgumentSyntax> arguments)
+        {
+            var builder = ImmutableArray.CreateBuilder<ExpressionSyntax>();
+            for (int i = 1; i < arguments.Count; i++)
+            {
+                builder.Add(CastAndParenthesize(arguments[i].Expression, semanticModel));
+            }
+
+            var expandedArguments = builder.ToImmutable();
+            return expandedArguments;
+        }
+
+        protected override LiteralExpressionSyntax GetFirstArgument(SeparatedSyntaxList<ArgumentSyntax> arguments) =>
+            arguments[0]?.Expression as LiteralExpressionSyntax;
+
+        protected override InterpolatedStringExpressionSyntax GetInterpolatedString(string text) =>
+            (InterpolatedStringExpressionSyntax)ParseExpression("$" + text);
```

Comment:
i'm wary of this blind cast.  I'd prefer an 'as' just in case the parser does something we don't expect.